azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/VirtualMachineApi.java

```
@@ -0,0 +1,140 @@
+@Consumes(MediaType.APPLICATION_JSON)
+public interface VirtualMachineApi {
+
+   /**
+    * The Get Virtual Machine details
+    */
+   @Named("GetVirtualMachine")
+   @GET
+   @Path("/{name}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   VirtualMachine get(@PathParam("name") String name);
+
+   /**
+    * The Get Virtual Machine details
+    */
+   @Named("GetVirtualMachineInstance")
+   @GET
+   @Path("/{name}/instanceView")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   VirtualMachineInstance getInstanceDetails(@PathParam("name") String name);
+
+   /**
+    * The Create Virtual Machine
+    */
+   @Named("CreateVirtualMachine")
+   @PUT
+   @Payload("%7B\"id\":\"{id}\",\"name\":\"{name}\",\"type\":\"Microsoft.Compute/virtualMachines\"," +
+           "\"location\":\"{location}\",\"tags\":%7B%7D,\"properties\":{properties}%7D")
+   @MapBinder(BindToJsonPayload.class)
+   @Path("/{vmname}")
+   @QueryParams(keys = "validating", values = "false")
+   VirtualMachine create(@PathParam("vmname") String vmname, @PayloadParam("id") String id,
+                         @PayloadParam("name") String name,
+                         @PayloadParam("location") String location,
+                         @PayloadParam("properties") VirtualMachineProperties properties);
+
+   /**
+    * The List Virtual Machines operation
+    */
+   @Named("ListVirtualMachines")
+   @GET
+   @SelectJson("value")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<VirtualMachine> list();
+
+   /**
+    * The Delete Virtual Machine operation
+    */
+   @Named("DeleteVirtualMachine")
+   @DELETE
+   @Path("/{name}")
+   @ResponseParser(URIParser.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   URI delete(@PathParam("name") String name);
+
+   /**
+    * The Restart Virtual Machine operation
+    */
+   @Named("RestartVirtualMachine")
+   @POST
+   @Path("/{name}/restart")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
```

We don't want to silently succeed if the VM does not exist. Remove the 404 fallbacks in all state change operations.